server/router/stats.ts

```diff
@@ -0,0 +1,155 @@
+import express, { Response } from 'express';
+import moment from 'moment-timezone';
+import { Condition } from 'dynamoose/dist/Condition';
+import { Stats } from '../models/stats';
+import { Ride } from '../models/ride';
+import * as db from './common';
+import { validateUser } from '../util';
+import { Driver, DriverType } from '../models/driver';
+
+const router = express.Router();
+const tableName = 'Stats';
+
+router.get('/:from/:to', validateUser('User'), async (req, res) => {
+  const { params: { from, to } } = req;
+
+  let date = from;
+  const dates = [];
+  while (date <= to) {
+    dates.push(date);
+    date = moment.tz(date, 'America/New_York').add(1, 'days').format('YYYY-MM-DD');
+  }
+
+  dates.forEach(async (currDate) => {
+    const year = moment.tz(currDate, 'YYYY-MM-DD', 'America/New_York').format('YYYY');
+    const monthDay = moment.tz(currDate, 'YYYY-MM-DD', 'America/New_York').format('MMDD');
+    const condition = new Condition()
+      .where('year')
+      .eq(year)
+      .and()
+      .where('monthDay')
+      .eq(monthDay);
+
+    const dateMoment = moment.tz(currDate, 'America/New_York');
+    // day = 12am to 5:00pm
+    const dayStart = dateMoment.toISOString();
+    const dayEnd = dateMoment.add(17, 'hours').toISOString();
+    // night = 5:01pm to 11:59:59pm
+    const nightStart = moment.tz(dayEnd, 'America/New_York').add(1, 'seconds').toISOString();
+    const nightEnd = moment.tz(currDate as string, 'America/New_York').endOf('day').toISOString();
+
+    computeStats(res, condition, dayStart, dayEnd, nightStart, nightEnd, year,
+      monthDay, currDate, to);
+  });
+});
+
+function computeStats(
+  res: Response,
+  condition: Condition,
+  dayStart: string,
+  dayEnd: string,
+  nightStart: string,
+  nightEnd: string,
+  year: string,
+  monthDay: string,
+  date: string,
+  to: string,
+) {
+  db.scan(res, Stats, condition, async (data: Document[]) => {
```

Comment:
since year and monthDay are keys, you dont need to do a scan

Reply:
I think I addressed all comments except this, bc I think in order to do this a GSI has to first be created that combines year and monthDay